src/transformers/models/splinter/modeling_splinter.py

```diff
@@ -0,0 +1,1556 @@
+            hidden_states=outputs.hidden_states,
+            attentions=outputs.attentions,
+        )
+
+
+def gather_positions(input_tensor, positions):
+    """
+    :param input_tensor: shape [batch_size, seq_length, dim] :param positions: shape [batch_size, num_positions]
+    :return: [batch_size, num_positions, dim]
+
+    Gathers specific vectors along the sequence dimension for classification
+    """
+    _, _, dim = input_tensor.size()
+    index = positions.unsqueeze(-1).repeat(1, 1, dim)  # [batch_size, num_positions, dim]
+    gathered_output = torch.gather(input_tensor, dim=1, index=index)  # [batch_size, num_positions, dim]
+    return gathered_output
+
+
+class FullyConnectedLayer(nn.Module):
+    def __init__(self, input_dim, output_dim, hidden_act="gelu"):
+        super(FullyConnectedLayer, self).__init__()
+
+        self.input_dim = input_dim
+        self.output_dim = output_dim
+
+        self.dense = nn.Linear(self.input_dim, self.output_dim)
+        self.act_fn = ACT2FN[hidden_act]
+        self.LayerNorm = nn.LayerNorm(self.output_dim)
+
+    def forward(self, inputs):
+        temp = self.dense(inputs)
+        temp = self.act_fn(temp)
+        temp = self.LayerNorm(temp)
+        return temp
+
+
+class QuestionAwareSpanSelectionHead(nn.Module):
+    """
+    Implementation of Question-Aware Span Selection (QASS) head, described in Splinter's paper:
+
+    """
+
+    def __init__(self, config):
+        super().__init__()
+
+        self.query_start_transform = FullyConnectedLayer(config.hidden_size, config.hidden_size)
+        self.query_end_transform = FullyConnectedLayer(config.hidden_size, config.hidden_size)
+        self.start_transform = FullyConnectedLayer(config.hidden_size, config.hidden_size)
+        self.end_transform = FullyConnectedLayer(config.hidden_size, config.hidden_size)
+
+        self.start_classifier = nn.Linear(config.hidden_size, config.hidden_size, bias=False)
+        self.end_classifier = nn.Linear(config.hidden_size, config.hidden_size, bias=False)
+
+    def forward(self, inputs, positions):
+        gathered_reps = gather_positions(inputs, positions)
+
+        query_start_reps = self.query_start_transform(gathered_reps)  # [batch_size, num_positions, dim]
+        query_end_reps = self.query_end_transform(gathered_reps)  # [batch_size, num_positions, dim]
+        start_reps = self.start_transform(inputs)  # [batch_size, seq_length, dim]
+        end_reps = self.end_transform(inputs)  # [batch_size, seq_length, dim]
+
+        temp = self.start_classifier(query_start_reps)  # [batch_size, num_positions, dim]
+        start_reps = start_reps.permute(0, 2, 1)  # [batch_size, dim, seq_length]
+        start_logits = torch.matmul(temp, start_reps)
+
+        temp = self.end_classifier(query_end_reps)
+        end_reps = end_reps.permute(0, 2, 1)
+        end_logits = torch.matmul(temp, end_reps)
+
+        return start_logits, end_logits
+
+
+class ClassificationHead(nn.Module):
```

Comment:
```suggestion
class SplinterClassificationHead(nn.Module):
```

Reply:
Actually this class is outdated so I removed it :)